gemini/use-cases/retrieval-augmented-generation/rag_evaluation/deepeval_with_gemini.ipynb

```diff
@@ -0,0 +1,840 @@
+    "### Create the evaluation set"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Evaluation set with questions and ground_truth\n",
+    "questions = [\n",
+    "    \"What architecture is proposed in paper titled Attention is all you need?\",\n",
+    "    \"Where do primary authors of paper titled Attention is all you need work?\",\n",
+    "]\n",
+    "ground_truth = [\"Transformers architecture\", \"Google Brain\"]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Run the [Q&A chain](#create-qa-chain) on evaluation dataset "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "contexts = []\n",
+    "answers = []\n",
+    "\n",
+    "# Generate contexts and answers for each question\n",
+    "for query in questions:\n",
+    "    result = qa({\"query\": query})\n",
+    "    contexts.append(\n",
+    "        [document.page_content for document in result.get(\"source_documents\")]\n",
+    "    )\n",
+    "    answers.append(result.get(\"result\"))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Convert into a dataset and prepare for consumption by DeepEval API\n",
+    "dataset = []\n",
+    "for q, a, c, g in itertools.zip_longest(questions, answers, contexts, ground_truth):\n",
+    "    dataset.append({\"Question\": q, \"Answer\": g, \"Context\": c})\n",
+    "\n",
+    "# Inspect the dataset\n",
+    "dataset"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## IMPORTANT : Gemini with DeepEval\n",
+    "> DeepEval is designed to work with OpenAI Models by default. We must write a Wrapper to make it work with Gemini"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Base LLM for Deepeval\n",
+    "\n",
+    "\n",
+    "class GoogleVertexAIDeepEval(DeepEvalBaseLLM):\n",
+    "    \"\"\"Class to implement Vertex AI for DeepEval\"\"\"\n",
+    "\n",
+    "    def __init__(self, model):  # pylint: disable=W0231\n",
+    "        self.model = model\n",
+    "\n",
+    "    def load_model(self):  # pylint: disable=W0221\n",
+    "        return self.model\n",
+    "\n",
+    "    def generate(self, prompt: str) -> str:  # pylint: disable=W0221\n",
+    "        chat_model = self.load_model()\n",
+    "        return chat_model.invoke(prompt).content\n",
+    "\n",
+    "    async def a_generate(self, prompt: str) -> str:  # pylint: disable=W0221\n",
+    "        chat_model = self.load_model()\n",
+    "        res = await chat_model.ainvoke(prompt)\n",
+    "        return res.content\n",
+    "\n",
+    "    def get_model_name(self):  # pylint: disable=W0236 , W0221\n",
+    "        return \"Vertex AI Model\""
```

Comment:
This looks like it's copied from the external python file. Would it be possible to download/import in the python file instead of copy-pasting it in? That can be removed once the Open Source Library is updated.

Reply:
@holtskinner This is as per documentation of DeepEval package : https://docs.confident-ai.com/docs/metrics-introduction#google-vertexai-example. It's deliberately kept in the notebook for users to understand that this needs to be implemented before they can use DeepEval with Gemini

Reply:
Ok, just make sure to remove it after the class is added to the DeepEval package, if relevant.